CorsixTH/Lua/app.lua

```diff
@@ -389,24 +389,38 @@ function App:init()
   return true
 end
 
---! Works out the intended location of the gamelog file.
+--! Works out the intended location of the gamelog file based on the time and save game name.
 --!return full path gamelog should exist at
-function App:getGamelogPath()
-  local config_path = self.command_line["config-file"] or ""
-  config_path = config_path:match("^(.-)[^" .. pathsep .. "]*$")
-  return config_path .. "gamelog.txt"
+function App:getGamelogPath(save_name)
+  return self.user_log_dir .. pathsep .. os.date("%y-%m-%d--%H-%M-%S--") ..
+      (save_name or 'unknown') .. ".txt"
 end
 
 --! Checks and creates the gamelog file if it does not exist.
-function App:initGamelogFile()
-  local gamelog_path = self:getGamelogPath()
+function App:initGamelogFile(save_name)
+  local gamelog_path = self:getGamelogPath(save_name)
   local gamelog = io.open(gamelog_path, "r")
   if gamelog then gamelog:close() return end
 
   local fi = self:writeToFileOrTmp(gamelog_path)
   local sysinfo = self:gamelogHeader()
   fi:write(sysinfo)
   fi:close()
+  self:trimLogs()
+end
+
+--! Trims the logs folder of old game logs down to ten files.
+function App:trimLogs()
+  local log_rentention, log_table = 11, {}
+  for node in lfs.dir(self.user_log_dir) do
+    local file = self.user_log_dir .. pathsep .. node
+    if lfs.attributes(file, "mode") == "file" then
+      table.insert(log_table, file)
+    end
```

Comment:
Probably not an issue, but we're deleting any file indiscriminately in this folder > 1mo.

Reply:
Unless you're trying to establish we just have write permission, toby?